controller/openebs/preinstallation.go

```diff
@@ -50,6 +51,22 @@ func (p *Planner) getPreInstallationManifests() error {
 			}
 			p.ComponentManifests[key] = value
 		}
+		// do not delete the existing components i.e., components which are already running in case of
+		// setting up ISCSI client.
+		// This will be the case where user wants to run ISCSI client setup when OpenEBS components
+		// are already running.
+		for _, component := range p.observedOpenEBSComponents {
+			if component.GetKind() == types.KindDaemonSet {
+				if component.GetName() == types.OpenEBSNodeSetupDaemonsetNameKey {
+					continue
+				}
+			} else if component.GetKind() == types.KindConfigMap {
+				if component.GetName() == types.OpenEBSNodeSetupConfigmapNameKey {
+					continue
```

Comment:
What is the logic behind "continue" ?

Reply:
We are not adding the two components that get created for the ISCSI client setup that is a daemonset and a configmap to the desired component list from this logic since they are already present in the desired components list.

Reply:
We are here only adding the components which were already present in the given namespace so that they don't get deleted.